app/main.py

```diff
@@ -32,13 +35,27 @@ def main() -> None:
     ]
 
     # run the programs
-    service.run_program(wake_up_program)
-    service.run_program(sleep_program)
+    await run_sequence(
+        run_parallel(
+            service.send_msg(wake_up_program[0]),
+            service.send_msg(wake_up_program[1])
+        ),
+        service.send_msg(wake_up_program[2])
+    )
+
+    await run_sequence(
+        run_parallel(
+            service.send_msg(sleep_program[0]),
+            service.send_msg(sleep_program[1]),
+            service.send_msg(sleep_program[2])
+        ),
+        service.send_msg(sleep_program[3])
```

Comment:
Similar to the previous comment, ensure that each call to `service.send_msg` is awaited within `run_parallel` and `run_sequence` to maintain the asynchronous flow.